lib/src/middleware.dart

```diff
@@ -48,17 +48,15 @@ Middleware createMiddleware(
   requestHandler ??= (request) async => request;
   responseHandler ??= (response) async => response;
 
-  return (inner) {
-    return new HandlerClient(
-      (request) => requestHandler(request)
-          .then((req) => inner.send(req))
-          .then((res) => responseHandler(res), onError: errorHandler),
-      onClose == null
-          ? inner.close
-          : () {
-              onClose();
-              inner.close();
-            },
-    );
-  };
+  return (inner) => new HandlerClient(
```

Comment:
I'm not convinced this `=>` change actually makes this more readable. There's already a *lot* of nesting and a lot of anonymous functions in the `HandlerClient` initialization, so I think having it clearly be a statement of its own makes this read more clearly.